cylc/uiserver/websockets/tornado.py

```diff
@@ -21,11 +22,13 @@
 
 setup_observable_extension()
 
+NO_MSG_DELAY = 1.0
+
 
 class TornadoConnectionContext(BaseConnectionContext):
     async def receive(self):
         try:
-            msg = await self.ws.recv()
+            msg = self.ws.recv_nowait()
             return msg
         except WebSocketClosedError:
             raise ConnectionClosedException()
```

Comment:
Shouldn't this not be a co-routine anymore? (It is now 'await'-less)

Reply:
(I guess it doesn't matter, is kind of pointless?)

Reply:
Oh, good point! It looks like it doesn't need to be `async` any more (or at least I don't see any reason). The function in the parent class is `def receive`, and implementations may use `async` if necessary I think.

Reply:
Done! :+1:  thanks!